src/worker/mod.rs

```diff
@@ -187,7 +192,16 @@ impl<S: AsyncBufRead + AsyncWrite + Send + Unpin, E: StdError + 'static + Send>
             .callbacks
             .get(job.kind())
             .ok_or(Failed::BadJobType(job.kind().to_string()))?;
-        handler.run(job).await.map_err(Failed::Application)
+        match handler {
+            Callback::Async(cb) => cb.run(job).await.map_err(Failed::Application),
+            Callback::Sync(cb) => {
+                let cb = Arc::clone(cb);
```

Comment:
Ah, this is why you `Arc` each individual `Sync` function. You could instead just `Arc::clone` all of `handler` I think to avoid the extra layer of `Arc`.

Reply:
Hmm, you mean having the sync handler boxed as advised [here](https://github.com/jonhoo/faktory-rs/pull/65#discussion_r1614433393) and  getting the `self.callbacks` (?) cloned instead? Because the individual handlers are not arced in the registry, while the registry itself is arced on the `Worker`. 

I might be missing something, please have a look:

```rust
async fn run_job(&mut self, job: Job) -> Result<(), Failed<E>> {
        let handler = self
            .callbacks
            .get(job.kind())
            .ok_or(Failed::BadJobType(job.kind().to_string()))?;
        match handler {
            Callback::Async(cb) => cb.run(job).await.map_err(Failed::Application),
            _ => {
                let callbacks = Arc::clone(&self.callbacks);
                spawn_blocking(move || {
                    if let Callback::Sync(cb) = callbacks
                        .get(job.kind())
                        .expect("sync handler to be available")
                    {
                        cb(job)
                    } else {
                        unreachable!();
                    }
                })
                .await
                .expect("joined ok")
                .map_err(Failed::Application)
            }
        }
    }
```
We are indeed avoiding introducing new arcs in this case, instead we cloning the whole of the registry. It just looks weird a little bit, wdyt?

Reply:
Yeah, that's exactly what I was thinking. I agree it looks pretty weird, and it's sad that we have to double-match. I think we should just stick with what we had with the double-`Arc`, even though it makes me sad.